File: src/libs/cpp/functions/mult.cpp
#include "mult.hpp"

void mult(Context &context) noexcept
{
	auto params = std::get<std::vector<Value>>(context.stack.pop());
	double total = 1;
	for (const Value &value : params)
	{
		if (std::holds_alternative<std::vector<Value>>(value))
		{
			for (const Value &inner_value : std::get<std::vector<Value>>(value))
			{
				total *= inner_value.to_number();
				if (total == 0)
				{
					break;
				}
			}
		}
		else
		{
			total *= value.to_number();
		}

		if (total == 0)
		{
			break;
		}
	}
	context.stack.push(total);
}

File: src/runtime/functions/mult.lua
return function(vm)
	local v, total = vm.pop(), 1
	for i = 1, #v do
		if type(v[i]) == 'table' then
			for k = 1, #v[i] do
				total = total * std.num(v[i][k])
				if total == 0 then break end
			end
		else
			total = total * std.num(v[i])
		end
		if total == 0 then break end
	end
	vm.push(total)
end
